module-4/backend/week_one.md

```diff
@@ -5,18 +5,59 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's the most useful thing you learned from completing the intermission week work?
+
+  I learned about unit testing with mocha and chai. 
+
 2. What are some tools to help debug JavaScript code?
+
+  The Chrome JavaScript developer tools w/debugger in the code, repl.it for JavaScript & pry for JavaScript.2. What are some tools to help debug JavaScript code?
+  
 3. What are some tools you need in order to unit test your JavaScript?
+
+  Mocha and Chai
+  
 4. What is the syntax for invoking a function?
+
+  functionName()
+  
 5. What is CORS?
+
+  CORS is Cross Origin Resource Sourcing which allows for cross-domain communication from the browser. It allows for domains to access data from each other.
+  
 6. How do we enable CORS?
+
+  CORS is enabled within access headers.
+  
 7. What's `this` in JavaScript?
+
+  'this' is a JS keyword which allows for a temporary value for the methods and properties of an object to be used in a function without needing to repeat the object.
+  
 8. What is Webpack and why is it useful?
+
+  Webpack is a JavaScript file bundler for the browser. It's especially useful for complex front end applications and handling file processing on a granular level allowing for more extensive control. 
+  
 9. When/why do you want to use event delegation?
-10. What is a callback function?
+
+  Event delegation allows for event listeners to be added to the parent element and bubbled to the child elements. This prevents repetition in the code and allows for child elements to be altered without needing to have event listeners added to them.
+  
+10. What's a callback function and what are some reasons when we use/need callback functions?
+
+  A callback function is a function that's passed as a parameter into another function and is executed within that function. Callback functions are best to use with AJAX calls because they ensure that a function will be run once a response is received.
+
 11. What's `npm` and what do we use it for?
 
+  npm is a JavaScript package manager that allows developers to reuse code created by other developers.
+
 #### Review  
-12. What's the MVC design pattern? Describe each part of MVC.
+
+12. What's the MVC design pattern? Describe each part of MVC?
+
+  The MVC design pattern stands for Model/View/Controller. It is a 3 part design pattern that handles the way in which an application functions. A controller handles the necessary actions which are activated by HTTP routes. The model is a representative of the data being accessed through the controller actions. The view handles the way in which the data is displayed to the client. 
+  
 13. What are a few ways to optimize a Rails application?
+
+  Rails applications can be optimized through caching, scaling and code optimization.
+
```

Comment:
Might be worth giving an example of something broad like "code optimization," such as minimizing DB queries.